index.html

```diff
@@ -122,6 +177,185 @@ <h2>Methods</h2>
+};
+    </pre>
+    <p>
+      Let [[\HeaderExtensionsToOffer]] be an internal slot of the RTCRtpTransceiver,
+      initialized to the platform-specific list of implemented RTP header extensions.
+      The direction
+      attribute for all extensions that are mandatory to use MUST be initialized to
+      an appropriate value other than "stopped". The direction attribute for extensions
+      that will not be offered by default in an initial offer MUST be initialized to
+      "stopped".
+    </p>
+    <p class="note">The list of header extensions that MUST/SHOULD be supported is listed in
+      [[RTCWEB-RTP]], section 5.2. The "mid" extension is mandatory to use when
+      BUNDLE is in use, per [[BUNDLE]] section 9.1.
+    </p>
+    <p>
+      Let [[\HeaderExtensionsNegotiated]] be an internal slot of the RTCRtpTransceiver,
+      initialized to an empty list.
+    </p>
+    <section>
+      <h2>Modifications to existing procedures</h2>
+      <p>
+        In the <a>set an RTCSessionDescription</a> algorithm, add a step
+        right after the step that sets transceiver.[[\Sender]].[[\SendCodecs]],
+        saying "For each transciever, set [[\HeaderExtensionsNegotiated]] to
+        an empty list, and then
+        for each "a=hdrext" line, add an appropriate
+        RTCRtpHeaderExtensionCapability to the list.
+      </p>
+      <p>
+        In the algorithms for generating initial offers in [[RTCWEB-JSEP]] section 5.2.1,
+        replace "for each supported RTP header extension, an "a=extmap" line, as specified in
+        [[RFC5285]], section 5" " with "For each RTP header extension "e"
+        listed in [[\HeaderExtensionsToOffer]] where direction is not "stopped", an "a=extmap"
+        line, as specified in [[RFC5285]], section 5, with direction taken from "e"'s direction
+        attribute."
+      </p>
+      <p>
+        In the algorithm for generating subsequent offers in [[RTCWEB-JSEP]] section 5.2.2, replace "The
+        RTP header extensions MUST only include those that are present in the most recent answer"
+        with "For each RTP header extension listed in [[\HeaderExtensionsToOffer]] that is
+        also present in the most recent answer, and where direction is not "stopped", generate
+        an appropriate "a=extmap" line with "direction" set according to the rules of [[RFC5285]]
+        section 6, considering the direction in [[\HeaderExtensionsToOffer]] to indicate the
+        answerer's desired usage". This preserves the property that the set of extensions
+        may not grow with subsequent offers.
+      </p>
+      <p>
+        In the algorithm for generating initial answers in [[RTCWEB-JSEP]] section 5.3.1, replace "For
+        each supported RTP header extension that is present in the offer" with "For each
+        supported RTP header extension that is present in the offer and is also present in
+        [[\HeaderExtensionsToOffer]] with a direction different from "stopped"".
```

Comment:
If the transceiver was created by setting the offer, [[\HeaderExtensionsToOffer]] would have its initial value of whatever the default is. Which means that if the offer contains a non-default header extension that we support, then by default we would reject it.

For transceiver that were created as a result of SRD(offer), might it make more more sense to initialize [[\HeaderExtensionsToOffer]] to the subset of what's on offer that we are capable of accepting?

This has several benefits:
1) The answerer does not have to figure out which header extensions to add in the most common case of accepting whatever is on offer.
2) This follows a similar pattern as transceiver.direction/currentDirection.
3) If between offer and answer, the answerer wants to know what was on offer, it can just check transceiver.headerExtensionsToOffer, because it reflects the offer. No SDP parsing needed.